### DISCUSSION TITLE
New Sheet Slot

### DISCUSSION THREAD

**Discussion by LucasAlbuquerque0** [Telefonica/mistica-design#1566] [Proposals]
### Goal

At BrandFactory we have the custom sheet that is used as a type of slot, where we can place an N number of different information. The objective is to bring the Sheet Slot (Custom Sheet) into figma.

### Context

In the request flows, some needs appeared where the use of the sheet slot is required. One of them is using Input Text within the component, guaranteeing the user the opportunity to quickly generate a comment or customize a service request request.

### Impact

The slot input will allow for more complete use of the component, allowing the use of new styles and components for an Action Sheet. Generating different single or even double actions.

### Discussion Questions

We already have it within BrandFactory, can we bring this Custom Sheet easily?

**Comment by yceballost** [comment]
Sure, we can include it in Figma but it is very important that Slot in sheets is only available for web implementations, **NOT for native implementation**

**Reply by LucasAlbuquerque0** [reply]
Would it be possible to expand to use natively?
A good part of the sheet's usage needs could be met with the use of a Native Custom Sheet.

**Reply by yceballost** [reply]
Not for Vivo Novo App, because Novo App **is a hybrid application**. The native component requires a bridge implementation to enable communication between native and web technologies, which is not compatible with the slot concept. The 'Bridge' requires specific parameters defined within the native component. The slot concept represents a 'free area,' conflicting with the requirements for bridge implementation.

**Reply by LucasAlbuquerque0** [reply]
I understand, free parameters can actually be problematic to implement depending on what is allocated within the slot.
So I will only keep the suggestions pre-defined by the rules that I sent the proposal to.

It would be very interesting if you could create a block for instance swap options 😂

**Reply by yceballost** [reply]
yea! we are open to include new types of sheets if the new types are thought in global terms! (avoiding specific needs for one product)

By the moment we have these types of sheet (that you can use with bridge):
- ActionSheet
- ActionListSheet
- InfoSheet
- RadioSheet
https://mistica-web.vercel.app/?path=/story/components-modals-sheet--actions

We don't really like this limitation with the sheet component because, for us, the sheet component should be just the "wrapped", I meant, the layer with opacity and the white container, and the content should be always free, like a common dialog. But when we are talking about hybrid technology, the things change T.T

If the Novo App were 100% native or 100% web, you could use sheet with slot

For example, this is a sheet with slot in web
https://jpeg.ly/P1I4i

**Comment by yceballost** [comment]
I think the dev team is looking for a solution to have webviews in fullscreen. if this possibility is carried out, we will can use custom sheets using web sheets inside a hybrid application like Meu Vivo (Novum) 

we are waiting to hear what happens with this from Novum side

**Reply by LucasAlbuquerque0** [reply]
Do you have any predictions on when the development team will bring this answer?

**Reply by yceballost** [reply]
no idea, this is something that is not part of Mística, but part of Novum's navigational framework. I can try to ask to somebody and i will notice you asap :)

**Comment by yceballost** [comment]
The problem with the Sheet is something to work in the Novum side, not from Mística. I will close this discussion.
